### PR TITLE
Audit logs

### PR DESCRIPTION
**Pull request by moelsayed**

### CODE REVIEW

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
++++
+title = "Audit Logging"
+date = 2020-04-02T14:50:15+02:00
+weight = 120
++++
+
+Audit Logging is one of key security features provided by Kubernetes. Once enabled in kube-api, it provides a chronological record of operations performed on the clusters by users, administrators and other cluster components.
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:7]: "cluster" in singular please. The feature has to be enabled for each cluster independently.

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Audit logging is also a key requirement of [Kubernetes CIS benchmark](https://www.cisecurity.org/benchmark/kubernetes/).
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:9]: requirement of *the*

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+For more details, you can can refer to the [upstream documentation](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/).
+
+{{% notice note %}}
+Kubernetes Audit Logging is introduced as a beta feature since **v1.11**.
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:14]: **was** introduced as a beta feature **in Kubernetes** v1.11.

Also, do we actually need this? We don't support anything <1.14.

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+{{% /notice %}}
+
+### Kubermatic Support
+Kubermatic provideds two levels of support for the Audit Logging:
+
+* Audit Logging on user-cluster level
+* Audit Logging on a Datacenter level
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:21]: lowercase "datacenter"

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+{{% notice note %}}
+Kubernetes Audit Logging is optional and is not enabled by default, since it requires additional memory and storage resources, depending on the specific configuration used.
+{{% /notice %}}
+
+Once enabled, Kubermatic will use a [Log Backend](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#log-backend) and a minimal [Policy](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy) by default:
+```yaml
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:28]: Add an empty line between the paragraph and the YAML snippet please.

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+{{< readfile "data/policy.yaml">}}
+```
+
+This file is stored in a ConfigMap named `audit-config` on the [Seed Cluster](../../concepts/architecture/#seed-cluster) in your cluster Namespace. To modify the default policy, you can edit this ConfigMap using `kubectl`:
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:32]: Please use a `({{< ref "../../concepts...." >}})`, so we notice breaking links while building the site.

Comment (xrstf) [reply content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:32]: Please lowercase "namespace".

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+```bash
+$ kubectl edit -n cluster-<YOUR CLUSTER ID> configmap audit-config
+```
+
+
+#### User-Cluster Level Audit Logging
+
+To enable user-cluster level Audit cluster, simply enable `Audit Logging` in the Kubermatic dashboard `Create Cluster` page:
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:41]: "Audit cluster" should probably be "Audit Logging"? Or are there words missing here?

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md (hunk 3 above)
Comment (moelsayed) [reply content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:14]: I am not sure. People are usually reluctant to upgrade. Are we sure we have no one running older versions ?

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+![Create Cluster](01-create-cluster.png)
+
+For exiting clusters, you can go to the cluster page, edit your cluster and enable (or disable) `Audit Logging`:
+
+![Edit Cluster](01-edit-cluster.png)
+
+#### Datacenter Level Audit Logging
+
+Kubermatic also supports enabling Audit Logging on the Datacenter level. In this case, the option is enforced on all user-clusters in the Datacenter. The user-cluster level flag is ignored in this case.
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:51]: Lowercase "datacenter".

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+To enable this, you will need to edit your [datacenters.yaml](../../concepts/datacenters/) or your [Seed Cluster CRD](../../concepts/seeds/).
```
Comment (xrstf) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:53]: datacenters.yaml is gone, so at the very least we should mention Seeds first and then, maybe, I would not, also include "... or edit your datacenters.yaml".

File: data/audit_log.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kubermatic.k8s.io/v1
```
Comment (xrstf) [inline data/audit_log.yaml:1]: Isn't this entire file redundant, given that `data/seed.yaml` exists (and is kept up-to-date, while the `audit_log.yaml` will be out-of-sync soon)?

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md (hunk 9 above)
Comment (moelsayed) [reply content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:53]: Only reason I included this is because the docs refer to datacenter.yaml as the main method and the CRD as an experimental option. Should I keep it ?

File: data/seed.yaml [hunk 11]
```diff
@@ -71,9 +71,6 @@ spec:
           # Datacenter location, e.g. "ams3". A list of existing datacenters can be found
           # at https://www.digitalocean.com/docs/platform/availability-matrix/
           region: ""
-        # EnforceAuditLogging enforces audit logging on every cluster within the DC,
-        # ignoring cluster-specific settings.
-        enforceAuditLogging: false
```
Comment (xrstf) [inline data/seed.yaml:76]: Why did you change this?

File: data/audit_log.yaml (hunk 10 above)
Comment (moelsayed) [reply data/audit_log.yaml:1]: seeds.yaml is very large, this is a simple snippet to show the basic structure.

File: data/seed.yaml (hunk 11 above)
Comment (moelsayed) [reply data/seed.yaml:76]: I was trying to make it more readable, didn't know it's auto generated The keys are sorted alphabetically, so some of the spec flags are listed between the cloud provider keys. will revert this.

File: data/audit_log.yaml (hunk 10 above)
Comment (xrstf) [reply data/audit_log.yaml:1]: But it will still get out-of-sync quickly. This example file is also not something a user can just copy-paste and apply straight-away to their cluster, so it's not a minimal viable example.

Comment (moelsayed) [reply data/audit_log.yaml:1]: Ok, will remove it.

File: content/kubermatic/master/tutorials/12-audit-logs/_index.en.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+* Audit Logging on a datacenter level
+
+{{% notice note %}}
+Kubernetes Audit Logging is optional and is not enabled by default, since it requires additional memory and storage resources, depending on the specific configuration used.
+{{% /notice %}}
+
+Once enabled, Kubermatic will use a [Log Backend](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#log-backend) and a minimal [Policy](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy) by default:
+
+```yaml
+{{< readfile "data/policy.yaml">}}
```
Comment (kron4eg) [inline content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:26]: I think this file is not there anymore. Please rebase the PR.

Comment (moelsayed) [reply content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:26]: @kron4eg, it is there. This PR adds it :+1:

Comment (kron4eg) [reply content/kubermatic/master/tutorials/12-audit-logs/_index.en.md:26]: Please move it out from `/data` dir, in hugo `/data` is interpreted by hugo (see https://gohugo.io/templates/data-templates/ for more info).

The appropriate place for it will be `/kubermatic/master/data`

Comment (moelsayed) [reply content/kubermatic/master/tutorials/12-audit-logs/_index.en.md]: @kron4eg Done. Thank you for clarifying!
